baseapp/files/rest_framework/uploads/views.py

```
@@ -0,0 +1,180 @@
+                user=request.user, profile=profile, **serializer.validated_data
+            )
+
+            # Prepare response
+            response_data = {"file_obj": file_obj, **upload_data}
+
+            response_serializer = UploadResponseSerializer(response_data)
+            return Response(response_serializer.data, status=status.HTTP_201_CREATED)
+
+        except ValueError:
+            logger.warning("Upload initiation rejected", exc_info=True)
+            raise ValidationError("Invalid upload parameters.")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)